lib/web/fetch/formdata.js

```diff
@@ -123,25 +121,23 @@ class FormData {
     const prefix = 'FormData.set'
     webidl.argumentLengthCheck(arguments, 2, prefix)
 
-    if (arguments.length === 3 && !(value instanceof Blob)) {
-      throw new TypeError(
-        "Failed to execute 'set' on 'FormData': parameter 2 is not of type 'Blob'"
-      )
+    name = webidl.converters.USVString(name)
+
+    if (arguments.length === 3 || value instanceof Blob) {
+      value = webidl.converters.Blob(value, prefix, 'value')
+
+      if (filename !== undefined) {
+        filename = webidl.converters.USVString(filename)
+      }
+    } else {
+      value = webidl.converters.USVString(value)
```

Comment:
cant this be at line 140?

Reply:
No. The webidl stuff occurs before the steps, but the step is also not needed because js doesn't have method overrides (hence the weird webidl here).

> The set(name, value) and set(name, blobValue, filename) method steps are:

The first step is telling us to set value to "blobValue" if there were 3 arguments passed, otherwise do nothing. "value" is only ever "value" here.